uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -13,19 +13,21 @@ define([
       return settings.pathTpl + 'components/manager-leave-calendar.html';
     }],
     controllerAs: 'calendar',
-    controller: ['$log', '$q', 'shared-settings', 'OptionGroup', 'Contact', 'AbsencePeriod', 'AbsenceType',
+    controller: ['$log', '$q', '$timeout', 'shared-settings', 'OptionGroup', 'Contact', 'AbsencePeriod', 'AbsenceType',
       'Calendar', 'LeaveRequest', 'PublicHoliday', controller]
   });
 
 
-  function controller($log, $q, sharedSettings, OptionGroup, Contact, AbsencePeriod, AbsenceType, Calendar, LeaveRequest, PublicHoliday) {
+  function controller($log, $q, $timeout, sharedSettings, OptionGroup, Contact, AbsencePeriod, AbsenceType, Calendar, LeaveRequest, PublicHoliday) {
     $log.debug('Component: manager-leave-calendar');
 
     var dayTypes = [],
       publicHolidays = [],
       leaveRequests = [],
+      leaveRequestsIndexed = [],
       leaveRequestStatuses = [],
-      vm = Object.create(this);
+      vm = Object.create(this),
+      tempContactData = [];
```

Comment:
Please refer point no 3 of description

> In loadCalendar instead of updating vm.managedContacts on completion of each contact's promise. Calendar data saved temporarily in tempContactData and once all the promises are resolved, data is transferred to vm.managedContacts. This is done so that browser paints the UI only once.